mix_pki/json.go

```diff
@@ -0,0 +1,173 @@
+// json.go - mixnet PKI client which uses json files
+// Copyright (C) 2017  David Anthony Stainton
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package provides mix PKI client implementations
+package mix_pki
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/katzenpost/core/crypto/ecdh"
+	"github.com/katzenpost/core/pki"
+	"github.com/katzenpost/core/sphinx/constants"
+	"github.com/op/go-logging"
+)
+
+var log = logging.MustGetLogger("mixclient")
+
+type JsonMixDescriptor struct {
```

Comment:
Why does this type need to be exported?

Can't this type be private? why not implement the https://golang.org/pkg/encoding/#TextMarshaler and TextUnmarshaler interfaces - by keeping the type private, renaming `JsonFromDescriptor` to `MarshalText`, and doing the JSON encoding of the non-exported jsonMixDescriptor that's created?